packages/victory-core/src/victory-state/victory-provider.tsx

```diff
@@ -69,18 +84,39 @@ export function VictoryProvider({
     };
   }, [props, domain, range]);
 
-  const value = {
-    scale,
-    data,
-    domain,
-    setChildProps,
-  };
+  const normalizedProps = React.useMemo(
+    () => ({
+      scale,
+      data,
+      domain,
+      updateChildProps,
+    }),
+    [scale, data, domain, updateChildProps],
+  );
+
+  return normalizedProps;
+}
+
+export function VictoryProvider({
+  children,
+  ...providerProps
+}: VictoryProviderProps) {
+  const value = useNormalizedProps();
```

Comment:
We will need some way of getting specific props for an individual component as well. Some will be the same, but some (like data or styles) might be specific to that component. So maybe we need to provide a `getProps` function through the provider value that accepts an ID?